src/modules/analysis/analysisEditContainer.js

```diff
@@ -0,0 +1,77 @@
+import inject from 'react-rxjs/dist/RxInject';
+import React from 'react';
+import analysisAddForm from './analysisAddForm';
+import AnalysisEditComponent from './AnalysisEditComponent';
+import store$, { loadAnalysis$, getAnalysisTypesForCollection$ } from './analysisStore';
+import Analysis from '../../models/analysis';
+import { makeUrlAware } from '../app/appSession';
+import flowRight from 'lodash/flowRight';
+import mount from '../../shared/mount';
+import { toPromise } from '../../shared/util';
+
+const { form$, updateForm$, loadForm$ } = analysisAddForm;
+
+const data = {
+  appSession$: { type: React.PropTypes.object.isRequired },
+  store$,
+  form$
+};
+
+const commands = {
+  updateForm$,
+  loadForm$,
+  getAnalysisTypesForCollection$,
+  loadAnalysis$
+};
+
+const props = {
+  editAnalysisEvent: toPromise(Analysis.editAnalysisEvent()),
+  loadAnalysisForForm: toPromise(Analysis.getAnalysisWithDeatils())
+};
+
+const callLoadAnalysisForForm = (loadAnalysisForForm, appSession, params, loadForm) =>
+  loadAnalysisForForm({
```

Comment:
you can make test for this by moving the .then part into the loadAnalysisForForm method.